test_scripts/Polices/External_UCS/External_Consent_OFF/014_ATF_Policies_External_Consent_OFF_disallowed_change_to_ON_allowed.lua

```diff
@@ -0,0 +1,175 @@
+  self.hmiConnection:SendNotification("SDL.OnAppPermissionConsent", {
+      source = "GUI",
+      externalConsentStatus = {{entityType = 2, entityID = 5, status = "OFF"}}
+    })
+  self.mobileSession:ExpectNotification("OnPermissionsChange")
+  :ValidIf(function(_,data)
+      local validate_result = common_functions_external_consent:ValidateHMIPermissions(data,
+        "SubscribeWayPoints", {allowed = {}, userDisallowed = {"BACKGROUND","FULL","LIMITED"}})
+      return validate_result
+    end)
+end
+
+--------------------------------------------------------------------------
+-- Precondition:
+-- RPC is disallowed to process.
+--------------------------------------------------------------------------
+Test["TEST_NAME_ON" .. "_Precondition_RPC_is_disallowed"] = function(self)
+  --mobile side: send SubscribeWayPoints request
+  self.mobileSession:SendRPC("SubscribeWayPoints",{})
+  --mobile side: SubscribeWayPoints response
+  EXPECT_RESPONSE("SubscribeWayPoints", {success = false , resultCode = "USER_DISALLOWED"})
+  EXPECT_NOTIFICATION("OnHashChange")
+  :Times(0)
+end
+
+--------------------------------------------------------------------------
+-- Main check:
+-- OnAppPermissionChanged is sent
+-- when HMI sends OnAppPermissionConsent with externalConsentStatus = ON
+--------------------------------------------------------------------------
+Test["TEST_NAME_OFF" .. "_MainCheck_HMI_sends_OnAppPermissionConsent_externalConsentStatus_ON"] = function(self)
+  --local hmi_app_id_1 = common_functions:GetHmiAppId(config.application1.registerAppInterfaceParams.appName, self)
+  -- hmi side: sending SDL.OnAppPermissionConsent for applications
+  self.hmiConnection:SendNotification("SDL.OnAppPermissionConsent", {
+      source = "GUI",
+      externalConsentStatus = {{entityType = 2, entityID = 5, status = "ON"}}
+    })
+  self.mobileSession:ExpectNotification("OnPermissionsChange")
+  :ValidIf(function(_,data)
+      local validate_result = common_functions_external_consent:ValidateHMIPermissions(data,
+        "SubscribeWayPoints", {allowed = {"BACKGROUND","FULL","LIMITED"}, userDisallowed = {}})
+      return validate_result
+    end)
+end
+
+--------------------------------------------------------------------------
+-- Main check:
+-- RPC is allowed to process.
+--------------------------------------------------------------------------
+Test["TEST_NAME_OFF" .. "_MainCheck_RPC_is_allowed"] = function(self)
+  --mobile side: send SubscribeWayPoints request
+  self.mobileSession:SendRPC("SubscribeWayPoints",{})
+  --hmi side: expected SubscribeWayPoints request
+  EXPECT_HMICALL("Navigation.SubscribeWayPoints")
+  :Do(function(_,data)
+      --hmi side: sending Navigation.SubscribeWayPoints response
+      self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS",{})
+    end)
+  --mobile side: SubscribeWayPoints response
+  EXPECT_RESPONSE("SubscribeWayPoints", {success = true , resultCode = "SUCCESS"})
```

Comment:
@istoimenova, Please update with usage of _correlationId_ because current version can be unstable